src/hooks/useDrag.tsx

```diff
@@ -595,121 +613,113 @@ export const onDelete = (
+        .hero as EditorHeroHighlightsSection).key_highlights,
+      indexToDelete
+    )
+    const newHighlightErrors = deleteElement(errors.highlights, indexToDelete)
+
+    const newDisplayHighlights = deleteElement(displayHighlights, indexToDelete)
+
+    return updateHighlightsSection(
+      homepageState,
+      newDisplayHighlights,
+      newHighlightOptions,
+      newHighlightErrors
+    )
+  }
+  if (elemType === "announcement") {
+    const announcementKeyExist = !_.isEmpty(
+      frontMatter.sections.find((section) =>
         EditorHomepageFrontmatterSection.isAnnouncements(section)
       )
-      const announcementsSection: AnnouncementsFrontmatterSection = frontMatter
-        .sections[announcementsIndex] as AnnouncementsFrontmatterSection
-
-      const newAnnouncementOptions = deleteElement(
-        announcementsSection.announcements.announcement_items,
-        indexToDelete
-      )
-      const newAnnouncementErrors = deleteElement(
-        errors.announcementItems,
-        indexToDelete
-      )
-
-      const newDisplayAnnouncements = deleteElement(
-        displayAnnouncementItems,
-        indexToDelete
-      )
-
-      return updateAnnouncementSection(
-        homepageState,
-        newDisplayAnnouncements,
-        newAnnouncementOptions,
-        newAnnouncementErrors,
-        announcementsIndex
-      )
-    }
-    case "textcardcard": {
-      const newTextCards = deleteElement(
-        ((frontMatter.sections[indexToDelete] as TextcardFrontmatterSection)
-          .textcards as EditorTextcardCardsSection).cards,
-        subindexToDelete
-      )
-      const newTextcardErrors = deleteElement(
-        errors.textcards[indexToDelete],
-        subindexToDelete
-      )
-      return updateTextCardsCardSection(
-        homepageState,
-        indexToDelete,
-        newTextCards,
-        newTextcardErrors
-      )
-    }
-    default:
+    )
+    if (!announcementKeyExist) {
+      // should not reach here, but defensively return the original state
       return homepageState
+    }
+
+    const announcementsIndex = frontMatter.sections.findIndex((section) =>
+      EditorHomepageFrontmatterSection.isAnnouncements(section)
+    )
+    const announcementsSection: AnnouncementsFrontmatterSection = frontMatter
+      .sections[announcementsIndex] as AnnouncementsFrontmatterSection
+
+    const newAnnouncementOptions = deleteElement(
+      announcementsSection.announcements.announcement_items,
+      indexToDelete
+    )
+    const newAnnouncementErrors = deleteElement(
+      errors.announcementItems,
+      indexToDelete
+    )
+
+    const newDisplayAnnouncements = deleteElement(
+      displayAnnouncementItems,
+      indexToDelete
+    )
+
+    return updateAnnouncementSection(
+      homepageState,
+      newDisplayAnnouncements,
+      newAnnouncementOptions,
+      newAnnouncementErrors,
+      announcementsIndex
+    )
+  }
+  if (elemType.startsWith("textcardcard")) {
```

Comment:
also need the check here

Reply:
Added type guard in https://github.com/isomerpages/isomercms-frontend/pull/1527/commits/96d0dddc2f7d06b599739083fad3887ffef41b9c